app/domain/identificationStatus.ts

```diff
@@ -36,14 +36,21 @@ export const needsToStartIdentification = (user: User) => {
   return !user.identified && !user.fscRequest;
 };
 
-export const showBundesidentPrimayOptionPage = (
-  bundesIdentIsOnline: boolean,
-  hasPrimaryOptionShown: boolean,
+export const hasPrimaryOptionEligibility = (
+  requestUrl: string,
+  serviceAvailability: boolean,
   user: User
 ) => {
+  const originQueryParam = new URL(requestUrl).searchParams.get("origin");
+  const isPrimaryOptionRequest = originQueryParam === "primaryoption";
+  const isDropoutSurveyRequest = originQueryParam === "survey";
+  const isBackButtonRequest = originQueryParam === "back";
+
   return (
-    bundesIdentIsOnline &&
-    !hasPrimaryOptionShown &&
+    !isPrimaryOptionRequest &&
+    !isDropoutSurveyRequest &&
+    !isBackButtonRequest &&
+    serviceAvailability &&
```

Comment:
Just out of curiosity: why did you decide for this approach compared to having specific routes?

Reply:
It's a more straightforward approach with less code change, this is the approach I wanted to tell you about, but because of my sickness, I forgot about it 😓 

which specific routes do you mean?

Reply:
Thanks, makes total sense 👍 I just meant linking directly to identifikation  primaryoption and survey and then only redirecting form the primary option if necessary.